source/daplink/interface/swd_host.c

```diff
@@ -878,6 +897,11 @@ uint8_t swd_set_target_state_hw(TARGET_RESET_STATE state)
             os_dly_wait(2);
             swd_set_target_reset(0);
             os_dly_wait(2);
+            
+            // Enable debug again because TARGET RESET might have disabled it
+            if (!swd_write_word(DBG_HCSR, DBGKEY | C_DEBUGEN)) {
+                return 0;
+            }
```

Comment:
Well, since its never used by the nRF51 nor the nRF52, this line is not necessary. It was added because the set_target_reset(0) might have disabled the debug connection. I can remove it.